spec/models/comment_spec.rb

```diff
@@ -456,9 +456,27 @@
   end
 
   context "when callbacks are triggered before save" do
-    it "generates character count before saving" do
-      comment.save
-      expect(comment.markdown_character_count).to eq(comment.body_markdown.size)
+    context "when the post is present" do
+      it "generates character count before saving" do
+        comment.save
+        expect(comment.markdown_character_count).to eq(comment.body_markdown.size)
+      end
+    end
+
+    context "when the commentable is not present" do
+      it "raises a validation error with message 'item has been deleted'" do
```

Comment:
Works great!

A tiny change: when we add multiple expectations we add `:aggregate_failures` to the `it` method.

```suggestion
      it "raises a validation error with message 'item has been deleted'", :aggregate_failures do
```